src/main/java/com/example/bookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,74 @@
+package com.example.bookstore.controller;
+
+import com.example.bookstore.dto.CategoryResponseDto;
+import com.example.bookstore.dto.CreateCategoryRequestDto;
+import com.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import com.example.bookstore.service.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/category")
+@RequiredArgsConstructor
+@Tag(name = "Categories", description = "API for managing categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @Operation(summary = "Create a new category (Only for admin)")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping
+    public CategoryResponseDto createCategory(@RequestBody CreateCategoryRequestDto requestDto) {
```

Comment:
You have validation annotations in DTO but don't have @Valid annotation in controller. For me it seems that validations will be skipped without it
Please recheck all occurences